src/main/java/org/fenixedu/academicextensions/services/registrationhistory/ConclusionReport.java

```diff
@@ -119,4 +119,12 @@ public static ConclusionReport empty() {
         return EMPTY_CONCLUSION;
     }
 
+    public String getConclusionNumber() {
+        if (conclusionBean == null) {
```

Comment:
Null check is not required in this case, because ConclusionReport requires conclusionbean (check constructor).
You can safely do:
return this.conclusionBean.getConclusionNumber();